plugin/src/main/resources/plugin.properties

```diff
@@ -1 +1 @@
-pluginClass=chisel3.internal.plugin.ChiselComponent
\ No newline at end of file
+pluginClass=chisel3.internal.plugin.ChiselPlugins
```

Comment:
Does this not cause problems for the Scala 2 plugin? Can you fork this into `src/main/resources-2` and `src/main/resources-3`?

Alternatively we could just keep the name `ChiselComponent`, but if you really want to call the Scala 3 version something else, that's fine, but the Scala 2 version has to work too.

Reply:
Scala 2 and Scala 3 plugin property files don't conflict cause Scala 3 expects a different name (plugin.properties) and in a different format from Scala 2's XML. All this is pretty poorly documented

I'm having issues splitting the resources into resources-2 and resources-3 - Scala 3 does not see the plugin.properties file when it's in `src/main/resources-3`. I thought I saw somewhere in Mill docs that this should work but I can't find it - do you have any pointers handy?

Reply:
> Scala 2 and Scala 3 plugin property files don't conflict cause Scala 3 expects a different name (plugin.properties) and in a different format from Scala 2's XML. All this is pretty poorly documented

Ah got it, thanks for the explanation.

But yeah we still want to split the resources since theirs no reason to ship the metadata for Scala 3 in the Scala 2 plugin and vice versa.

> I'm having issues splitting the resources into resources-2 and resources-3 

This split is not a property of Scala itself, it's something we'll have to do ourselves in Mill. It seems that it does such a split automatically for sources but not for resources. Basically in `plugin/package.mill` we should `override def resources` and have it use the Scala major version in the resource directories.

Reply:
Handled here: #4971